src/mqtt-client/client.js

```diff
@@ -59,6 +72,16 @@ function onSubscribed() {
             deviceName = topic.split('/').pop();
 
         if (isDeviceInfo) {
+            debug(`Device-info came: ${deviceName}, publishingQueue.size: ${publishingQueue.size}`);
```

Comment:
Regarding timeout I agree. About extract to separate file - I thought a lot about it but I will leave it for refactoring also...